iree/turbine/kernel/wave/scheduling/schedule.py

```diff
@@ -112,7 +118,12 @@ def schedule_reduction(
     )
 
     # Update new reduction count.
-    new_reduction.count = max_induction_variable - (scheduler.num_stages - 1)
+    if ivar_is_number:
+        new_reduction.count = max_induction_variable - (scheduler.num_stages - 1)
+    else:
+        new_reduction.count = sympy.Max(
+            0, max_induction_variable - (scheduler.num_stages - 1)
```

Comment:
Just curious, why wouldn't the count be 1 instead of 0?

Reply:
Because this is to handle situations where we don't have enough iterations (count < 0), but based on our discussion I will add an Assumption to handle this.